Вариант 1:
	try {
		начать работу
		работать
	} catch (e) {
		обработать ошибку
	} finally {
		очистить рабочее пространство
	}

Вариант 2:
	try {
		начать работу
		работать
	} catch (e) {
		обработать ошибку
	}

	очистить рабочее пространство


Разница между двумя этими вариантами есть!
Преимущество первго варианта в том, что очистка рабочего пространства
произойдет даже в том случае, если блок catch не сможет обработать ошибку
и прокинет ее дальше

А во втором варианте, при возникновении глобальной ошибке (которую не знает cathc),
очистка рабочего пространства не произойдет